packages/fx-core/src/plugins/resource/appstudio/plugin.ts

```diff
@@ -164,35 +164,43 @@ export class AppStudioPluginImpl {
   ): Promise<Result<string, FxError>> {
     const appDirectory = await getAppDirectory(ctx.root);
     const appStudioToken = await ctx.appStudioToken?.getAccessToken();
-    const componentID = manifest.id;
+    const localTeamsAppID = this.getTeamsAppId(ctx, true);
+    let createIfNotExist = false;
+    if (localTeamsAppID) {
+      manifest.id = localTeamsAppID;
+    } else {
+      manifest.id = "";
+      createIfNotExist = true;
+    }
     if (manifest.configurableTabs) {
       for (const tab of manifest.configurableTabs) {
-        tab.configurationUrl = `https://{teamSiteDomain}{teamSitePath}/_layouts/15/TeamsLogon.aspx?SPFX=true&dest={teamSitePath}/_layouts/15/TeamsWorkBench.aspx%3FcomponentId=${componentID}%26openPropertyPane=true%26teams%26forceLocale={locale}%26loadSPFX%3Dtrue%26debugManifestsFile%3Dhttps%3A%2F%2Flocalhost%3A4321%2Ftemp%2Fmanifests.js`;
+        const reg = /[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}/;
+        const result = tab.configurationUrl.match(reg);
```

Comment:
what if the reg match fail? will it throw error?

Reply:
thanks, add log and warning message.
![image](https://user-images.githubusercontent.com/71362691/133017154-561d67e0-7505-4597-b3a5-2e3c836b3c13.png)